manifests/sysctl.pp

```diff
@@ -0,0 +1,250 @@
+        'net.ipv4.conf.default.send_redirects':       value => $net__ipv4__conf__default__send_redirects;
+        'net.ipv4.icmp_echo_ignore_broadcasts':       value => $net__ipv4__icmp_echo_ignore_broadcasts;
+        'net.ipv4.icmp_ignore_bogus_error_responses': value => $net__ipv4__icmp_ignore_bogus_error_responses;
+        'net.ipv4.tcp_challenge_ack_limit':           value => $net__ipv4__tcp_challenge_ack_limit;
+        'net.ipv4.tcp_max_syn_backlog':               value => $net__ipv4__tcp_max_syn_backlog;
+        'net.ipv4.tcp_syncookies':                    value => $net__ipv4__tcp_syncookies;
+      }
+
+      file { '/var/core':
+        ensure => 'directory',
+        owner  => 'root',
+        group  => 'root',
+        mode   => '0640',
+        before => [
+          Sysctl['kernel.core_pattern'],
+          Sysctl['kernel.core_uses_pid'],
+        ]
+      }
+      if !$core_dumps {
+        pam::limits::add { 'prevent_core':
+          domain => '*',
+          type   => 'hard',
+          item   => 'core',
+          value  => '0',
+          order  => '100'
+        }
+      }
+
+      if ( $::operatingsystem in ['RedHat','CentOS'] ) and ( $::operatingsystemmajrelease == '6' ) {
+        sysctl { 'kernel.exec-shield': value => $kernel__exec_shield; }
+      }
+
+      if $enable_ipv6 {
+        sysctl { 'net.ipv6.conf.all.disable_ipv6': value => '0'; }
+          ->
+        sysctl {
+          'net.ipv6.conf.all.accept_redirects':         value => $net__ipv6__conf__all__accept_redirects;
+          'net.ipv6.conf.all.autoconf':                 value => $net__ipv6__conf__all__autoconf;
+          'net.ipv6.conf.all.forwarding':               value => $net__ipv6__conf__all__forwarding;
+          'net.ipv6.conf.default.accept_ra':            value => $net__ipv6__conf__default__accept_ra;
+          'net.ipv6.conf.default.accept_ra_defrtr':     value => $net__ipv6__conf__default__accept_ra_defrtr;
+          'net.ipv6.conf.default.accept_ra_pinfo':      value => $net__ipv6__conf__default__accept_ra_pinfo;
+          'net.ipv6.conf.default.accept_ra_rtr_pref':   value => $net__ipv6__conf__default__accept_ra_rtr_pref;
+          'net.ipv6.conf.default.accept_redirects':     value => $net__ipv6__conf__default__accept_redirects;
+          'net.ipv6.conf.default.autoconf':             value => $net__ipv6__conf__default__autoconf;
+          'net.ipv6.conf.default.dad_transmits':        value => $net__ipv6__conf__default__dad_transmits;
+          'net.ipv6.conf.default.max_addresses':        value => $net__ipv6__conf__default__max_addresses;
+          'net.ipv6.conf.default.router_solicitations': value => $net__ipv6__conf__default__router_solicitations;
+        }
+      }
+      else {
+        if $facts['ipv6_enabled'] {
+          sysctl { 'net.ipv6.conf.all.disable_ipv6': value => '1'; }
+        }
+      }
+    }
+    default : {
+      warning('Only RedHat and CentOS are currently supported by "simp::sysctl"')
```

Comment:
Hmm...should this be a warning or a failure?

It was a warning before because it was stuffed in simplib and bundled. It's separate now.